connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -382,6 +382,26 @@ public static SchemaBuilder map(Schema keySchema, Schema valueSchema) {
         return builder;
     }
 
+    static SchemaBuilder arrayOfNull() {
```

Comment:
This part is slightly hacky since the `SchemaBuilder` is part of the public API. You'll notice the additional methods here are package-private, which is intentional in order to avoid altering the public API. As a result, I also think the method names matter slightly less than they might normally, and the additional `mapWithNullKeys` and `mapWithNullValues` methods are better left as separate methods instead of folding that functionality into the existing `map` method.
Also, in the event that we want to add an `array()` or a `map()` method as part of the public API in the future that does something else, it seems better to make these internal methods more specific so as to be less likely to pollute that namespace.